python/pyiceberg/catalog/hive.py

```diff
@@ -197,17 +234,41 @@ def __init__(self, name: str, properties: Properties, uri: str):
         super().__init__(name, properties)
         self._client = _HiveClient(uri)
 
-    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
-        # Requires reading the manifest, will implement this in another PR
-        # Check the table type
-        return Table(identifier=(table.dbName, table.tableName))
+    def _convert_hive_into_iceberg(self, table: HiveTable, metadata_location: Optional[str] = None) -> Table:
+        properties: Dict[str, str] = table.parameters
+        if TABLE_TYPE not in properties or properties[TABLE_TYPE] != ICEBERG:
+            raise NoSuchTableError(f"Table is not an Iceberg table: {table.dbName}.{table.tableName}")
+
+        if not metadata_location:
+            if prop_metadata_location := properties.get(METADATA_LOCATION):
+                metadata_location = prop_metadata_location
+            else:
+                raise NoSuchTableError("Metadata location not found")
+
+        file = PyArrowFile(metadata_location)
```

Comment:
File implementations should not be used directly. Instead, this should have a `FileIO` instance passed in to use for reading:

```
file = io.newInputFile(metadata_location)
```

Reply:
I didn't want to spend too much time on it in this PR. I've created a stub for it 👍🏻